sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -82,17 +82,94 @@ class BearerTokenCredentialPolicy(_BearerTokenCredentialPolicyBase, SansIOHTTPPo
 
     def on_request(self, request):
         # type: (PipelineRequest) -> None
-        """Adds a bearer token Authorization header to request and sends request to next policy.
+        """Called before the policy sends a request.
 
-        :param request: The pipeline request object
-        :type request: ~azure.core.pipeline.PipelineRequest
+        The base implementation authorizes the request with a bearer token.
+
+        :param ~azure.core.pipeline.PipelineRequest request: the request
         """
         self._enforce_https(request)
```

Comment:
How do you think about moving _enforce_https into shared utils?

Reply:
Makes sense the next time another policy needs it. Do we want the key and SAS policies to enforce HTTPS as well?

Reply:
I asked this because I saw ACR also used it. :)